operators/src/adapters/raster_overlap_adapter.rs

```diff
@@ -0,0 +1,528 @@
+where
+    T: Pixel,
+    Q: RasterQueryProcessor<RasterType = T>,
+    QC: QueryContext,
+    XF: Clone + Fn(TileInformation, QueryRectangle) -> Result<(RasterTile2D<T>, X)>,
+    X: Clone + Send,
+
+    FoldFut: Future<Output = Result<(RasterTile2D<T>, X)>>,
+    FoldF: Clone + Fn(Result<(RasterTile2D<T>, X)>, Result<RasterTile2D<T>>) -> FoldFut,
+
+    QF: Clone + Fn(&TileInformation, &QueryRectangle, TimeInstance) -> Result<QueryRectangle>,
+{
+    fn is_terminated(&self) -> bool {
+        self.ended
+    }
+}
+
+impl<'a, T, Q, QC, FoldF, FoldFut, XF, X, QF> Stream
+    for RasterOverlapAdapter<'a, T, Q, QC, FoldF, FoldFut, XF, X, QF>
+where
+    T: Pixel,
+    Q: RasterQueryProcessor<RasterType = T>,
+    QC: QueryContext,
+    XF: Clone + Fn(TileInformation, QueryRectangle) -> Result<(RasterTile2D<T>, X)>,
+    X: Clone + Send,
+
+    FoldFut: Future<Output = Result<(RasterTile2D<T>, X)>>,
+    FoldF: Clone + Fn(Result<(RasterTile2D<T>, X)>, Result<RasterTile2D<T>>) -> FoldFut,
+
+    QF: Clone + Fn(&TileInformation, &QueryRectangle, TimeInstance) -> Result<QueryRectangle>,
+{
+    type Item = Result<RasterTile2D<T>>;
+
+    fn poll_next(
+        self: Pin<&mut Self>,
+        cx: &mut std::task::Context<'_>,
+    ) -> std::task::Poll<Option<Self::Item>> {
+        if self.is_terminated() {
+            return Poll::Ready(None);
+        }
+
+        let mut this = self.project();
+
+        if this.running_future.as_ref().is_none() {
+            if *this.current_spatial_tile >= this.spatial_tiles.len() {
+                *this.current_spatial_tile = 0;
+                *this.time_start = *this.time_end;
+                *this.time_end = None;
+            }
+
+            if let Some(t_start) = *this.time_start {
+                if t_start >= this.query_rect.time_interval.end() {
+                    *this.ended = true;
+                } else {
+                    // TODO: maybe this whole block should also be a future!
+                    let current_spatial_tile_info = this.spatial_tiles[*this.current_spatial_tile];
+
+                    let fold_tile_spec = TileInformation {
+                        tile_size_in_pixels: current_spatial_tile_info.tile_size_in_pixels,
+                        global_tile_position: current_spatial_tile_info.global_tile_position,
+                        global_geo_transform: current_spatial_tile_info.global_geo_transform,
+                    };
+
+                    let tqr = (this.tile_query_rewrite)(
+                        &current_spatial_tile_info,
+                        &this.query_rect,
+                        t_start,
+                    )?;
+
+                    let qs = this.source.raster_query(tqr, *this.query_ctx)?;
+
+                    let accu = (this.accu_creator)(fold_tile_spec, tqr);
+
+                    let ttf = qs.fold(accu, this.fold_fn.clone());
+
+                    this.running_future.set(Some(ttf));
```

Comment:
Where do you start a thread to compute it?

Reply:
in the fold method